### PR TITLE
one way to deprecate them all

### PR DESCRIPTION
**Pull request by bcoca**
consolidate all variations into one object

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

deprecation

### CODE REVIEW

**Comment by mkrizek** [comment]
!needs_collection_redirect

**Comment by mkrizek** [comment]
!needs_collection_redirect

**Comment by bcoca** [comment]
!bot_ignore

File: lib/ansible/module_utils/common/deprecations.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        if self.collection_name == 'ansible.builtin':
+            self.collection_name = 'ansible-core'
+
+    def __str__(self):
+
+        msg = '{0}  This feature will be removed from {1}'.format(self.why, self.collection_name)
+
+        if self.when is None:
+            msg += 'in a future release.'
+        if date_string.match(self.when):
```
Comment (flowerysong) [inline lib/ansible/module_utils/common/deprecations.py]:
```suggestion
        elif date_string.match(self.when):
```

**Pull request closed without merging**